### PR TITLE
[BIOMAGE-1921] - Decouple worker pod configs

### PR DESCRIPTION
**Pull request by aerlaut**
# Description
Decouple worker ci config

# Details
#### URL to issue
https://biomage.atlassian.net/browse/BIOMAGE-1921

https://ui-agi-cloudy-dragonfly.scp-staging.biomage.net/
#### Link to staging deployment URL (or set N/A)
N/A

#### Integration test branch
master

# Merge checklist
Your changes will be ready for merging after all of the steps below have been completed.

### Code updates
Have best practices and ongoing refactors being observed in this PR
- [ ] Migrated any selector / reducer used to the new format.

### Manual/unit testing
- [ ] Tested changes using InfraMock locally **or** no tests required for change, e.g. Kubernetes chart updates.
- [ ] Validated that current unit tests for code work as expected and are sufficient for code coverage **or** no unit tests required for change, e.g. documentation update.
- [ ] Unit tests written **or** no unit tests required for change, e.g. documentation update.

### Integration testing
**You must check the box below** to run integration tests on the latest commit on your PR branch.
Integration tests have to pass before the PR can be merged. Without checking the box, your PR
**will not pass** the required status checks for merging.

- [ ] Started end-to-end tests on the latest commit.

### Documentation updates
- [ ] Relevant Github READMEs updated **or** no GitHub README updates required.
- [ ] Relevant Wiki pages created/updated **or** no Wiki updates required.

### Optional
- [ ] Staging environment is unstaged before merging.
- [ ] Photo of a cute animal attached to this PR.

### CODE REVIEW

**Review by StefanBabukov — APPROVED** [review]
lgtm

**Comment by ivababukova** [comment]
@aerlaut can you make sure you add AWS_REGION and AWS_ACCOUNT_ID as environment variables to the pod? If you don't, then this piece of code in the worker config:

```
aws_account_id = os.getenv("AWS_ACCOUNT_ID", default="242905224710")
aws_region = os.getenv("AWS_DEFAULT_REGION", default="eu-west-1")
```
will make the worker account-specific. Also, if you could modify the aws_region variable to use `AWS_REGION` instead and remove those defaults, raise an error if they are not defined.

File: chart-infra/templates/pod-template.tpl [hunk 1]
```diff
@@ -7,6 +7,11 @@
       containers:
       - name: "{{ .Release.Name }}-r"
         image: "{{ .Values.r.image }}"
+        env:
```
Comment (ivababukova) [inline chart-infra/templates/pod-template.tpl]: typo in xperiments in the beginning of the file

File: chart-infra/templates/pod-template.tpl [hunk 2]
```diff
@@ -25,6 +30,10 @@
       - name: "{{ .Release.Name }}"
         image: "{{ .Values.python.image }}"
         env:
+        - name: AWS_ACCOUNT_ID
+          value: "{{ .Values.myAccount.accountId }}"
+        - name: AWS_REGION
+          value: "{{ .Values.myAccount.region }}"
```
Comment (ivababukova) [inline chart-infra/templates/pod-template.tpl]: can you remove region from here?

File: python/src/worker/config/__init__.py [hunk 3]
```diff
@@ -113,6 +112,7 @@ def REDIS_CLIENT(self):
     TIMEOUT=timeout,
     IGNORE_TIMEOUT=ignore_timeout,
     AWS_ACCOUNT_ID=aws_account_id,
+    AWS_DEFAULT_REGION=aws_region,
     AWS_REGION=aws_region,
```
Comment (ivababukova) [inline python/src/worker/config/__init__.py]: can you remove AWS_REGION?

File: python/src/worker/config/__init__.py [hunk 4]
```diff
@@ -131,8 +131,11 @@ def REDIS_CLIENT(self):
 
 if cluster_env == "development" or cluster_env == "test":
     config.AWS_ACCOUNT_ID = "000000000000"
+    config.AWS_DEFAULT_REGION = "eu-west-1"
+    config.AWS_REGION = "eu-west-1"
```
Comment (ivababukova) [inline python/src/worker/config/__init__.py]: can you remove AWS_REGION?

File: chart-infra/templates/pod-template.tpl [hunk 5]
```diff
@@ -7,6 +7,11 @@
       containers:
       - name: "{{ .Release.Name }}-r"
         image: "{{ .Values.r.image }}"
+        env:
+        - name: AWS_ACCOUNT_ID
+          value: "{{ .Values.myAccount.accountId }}"
+        - name: AWS_DEFAULT_REGION
+          value: "{{ .Values.myAccount.region }}"
```
Comment (ivababukova) [inline chart-infra/templates/pod-template.tpl]: check if they are used in r and if they are not, don't add them

File: python/src/worker/config/__init__.py [hunk 6]
```diff
@@ -143,3 +146,6 @@ def REDIS_CLIENT(self):
 
 if cluster_env != "test":
     core.patch_all()
+
+if not config.AWS_ACCOUNT_ID or not config.AWS_DEFAULT_REGION   :
+    raise Exception("env.AWS_ACCOUNT_ID or env.AWS_DEFAULT_REGION is not set")
```
Comment (ivababukova) [inline python/src/worker/config/__init__.py]: split them up

File: chart-infra/templates/pod-template.tpl [hunk 7]
```diff
@@ -7,6 +7,7 @@
       containers:
       - name: "{{ .Release.Name }}-r"
         image: "{{ .Values.r.image }}"
+        env:
```
Comment (ivababukova) [inline chart-infra/templates/pod-template.tpl]: remove this env

**Review by ivababukova — APPROVED** [review]

**Pull request merged**